test/test_extensions.py

```diff
@@ -1,25 +1,64 @@
+import inspect
+
 import pytest
+from sqlalchemy import orm
 
 import psqlgraph
 from psqlgraph import ext
 
+SNode, SEdge = ext.register_base_class("sample")
+SBase = ext.get_orm_base("sample")
+
+TNode, TEdge = ext.register_base_class("test")
+TBase = ext.get_orm_base("test")
+
+globals()["SNode"] = SNode
+
+
+def foo(node: SNode) -> int: ...
+
 
 @pytest.mark.parametrize(
-    "ns, node_cls_name, edge_cls_name",
-    [
-        ("sample", "SampleAbstractNode", "SampleAbstractEdge"),
-        ("test", "TestAbstractNode", "TestAbstractEdge"),
-        (None, "Node", "Edge"),
-    ],
+    "ns",
+    ("sample", "test", None),
 )
-def test_register_bases(ns, node_cls_name, edge_cls_name):
+def test_register_bases(ns):
 
     node_cls, edge_cls = ext.register_base_class(package_namespace=ns)
     assert issubclass(node_cls, psqlgraph.AbstractNode)
     assert issubclass(edge_cls, psqlgraph.AbstractEdge)
 
-    assert node_cls.__name__ == node_cls_name
-    assert edge_cls.__name__ == edge_cls_name
+
+def test_base_classes_distinct():
+    def create_node(node_base: type[psqlgraph.AbstractNode]) -> type[psqlgraph.AbstractNode]:
+        class AlignedReads(node_base):
+            __label__ = "aligned_reads"
+            __tablename__ = "node_aligned_reads"
+
+            @psqlgraph.pg_property(float)
+            def height(self, value):
+                self._set_property("height", value)
+
+        # Insure that configuration can be called for both.
+        orm.configure_mappers()
+
+        return AlignedReads
+
+    # Check that the base classes are not the same.
+    assert SNode != TNode
+    assert SEdge != TEdge
+
+    sample = create_node(SNode)
+    test = create_node(TNode)
+
+    # Insure that despite same name classes are not same & that they are not
+    # contaminated with the opposites base class.
+    assert sample.__name__ == test.__name__
+    assert sample != test
+    assert TNode not in inspect.getmro(sample)
+    assert SNode in inspect.getmro(sample)
+    assert SNode not in inspect.getmro(test)
+    assert TNode in inspect.getmro(test)
```

Comment:
@kulgan This is the test I added to try and recreate the scenario that I think you were describing but despite the same `__qualname__` these base classes and even derived classes with the same name seem to be quite distinct. Did I miss a nuance to what you were trying to say tho?